Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py

```diff
@@ -194,6 +215,32 @@ def pickable(cls, commits, repository, commits_story=None, excluded=None):
 
         return [commit for commit in commits if str(commit) in filtered_in]
 
+    @classmethod
+    def related_commits_from_notes(cls, commit, run_command_function=None):
+        '''
+        Parse Git notes for patterns like:
+            Picked-from: abc1234
+            Cherry-picked-from: <hash>
+        '''
+
+        # If no custom run_command_function is provided for mocking, fall back to the default `run` method
+        run_command_function = run_command_function or run
+
+        result = run_command_function(['git', 'notes', 'show', commit.hash], capture_output=True)
+        if result.returncode != 0:
+            return []
+
+        # Decode stdout from bytes to string if it's in bytes format
+        stdout_str = result.stdout.decode('utf-8') if isinstance(result.stdout, bytes) else result.stdout
+
+        related = []
+        for line in stdout_str.strip().splitlines():
+            match = re.search(r'(Cherry-picked|Picked)-from:\s+([0-9a-f]{6,40})', line, re.IGNORECASE)
```

Comment:
This is a pretty specific regex. First, it needs to support identifiers (there are a number of regexes that do both that you can copy, maybe even use directly). Second, because this is so specific, we probably need a `git-webkit skip` or `git-webkit annotate` that helps users craft such a note